src/js/components/modals/UpdateConfigModal.js

```diff
@@ -0,0 +1,255 @@
+import classNames from 'classnames';
+import {Modal} from 'reactjs-components';
+import mixin from 'reactjs-mixin';
+import React from 'react';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import AdvancedConfig from '../AdvancedConfig';
+import CosmosErrorMessage from '../CosmosErrorMessage';
+import CosmosPackagesStore from '../../stores/CosmosPackagesStore';
+import Service from '../../structs/Service';
+import StringUtil from '../../utils/StringUtil';
+import UniversePackage from '../../structs/UniversePackage';
+import Util from '../../utils/Util';
+
+const METHODS_TO_BIND = [
+  'handleAdvancedFormChange',
+  'handleConfigSave',
+  'handleErrorConfigEditClick',
+  'handleModalClose',
+  'getConfigSubmit'
+];
+
+class UpdateConfigModal extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      hasFormErrors: false,
+      updateError: null,
+      pendingRequest: false,
+      updateSuccess: false
+    };
+
+    this.store_listeners = [{
+      name: 'cosmosPackages',
+      events: [
+        'installError',
+        'installSuccess'
+      ]
+    }];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onCosmosPackagesStoreInstallError(error) {
+    this.setState({
+      updateError: error,
+      pendingRequest: false,
+      updateSuccess: false
+    });
+  }
+
+  onCosmosPackagesStoreInstallSuccess() {
+    this.setState({
+      updateError: null,
+      pendingRequest: false,
+      updateSuccess: true
+    });
+  }
+
+  handleAdvancedFormChange(formData) {
+    this.setState({hasFormErrors: !formData.isValidated});
+  }
+
+  handleConfigSave() {
+    let serviceMetadata = this.props.service.getMetadata();
+    let {name, version} = serviceMetadata;
+
+    let {isValidated, model} = this.triggetConfigSubmit();
+
+    if (isValidated) {
+      CosmosPackagesStore.installPackage(name, version, model);
+      this.setState({pendingRequest: true});
+    }
+  }
+
+  handleErrorConfigEditClick() {
+    this.setState({updateError: null});
+  }
+
+  handleModalClose() {
+    this.setState({
+      pendingRequest: false,
+      updateError: null,
+      updateSuccess: false
+    });
+    this.props.onClose();
+  }
+
+  getConfigSubmit(triggerSubmit) {
+    this.triggetConfigSubmit = triggerSubmit;
+  }
+
+  getUpdateErrorContent() {
+    let {updateError} = this.state;
+
+    return (
+      <CosmosErrorMessage
+        className="text-error-state text-overflow-break-word text-align-center"
+        error={updateError}
+        wrapperClass="modal-content-inner container container-pod
+          container-pod-short horizontal-center" />
+    );
+  }
+
+  getModalContents() {
```

Comment:
Sure, but what is the logic for this? I guess I'm just not sure how to order these methods